TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -90,45 +91,40 @@ class BenchmarkParser extends Parser {
             }
 
             if ( isValid ) {
-
-                curBenchmarkNamedRouted = BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant];
+            	
+                curBench = BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]];
 
                 // Parse metric values
                 curTestData["metrics"] = [];
-                for ( let i = 0; i < BenchmarkMetric[curBenchmarkNamedRouted]["metrics"].length; i++ ) {
-                    curMetric = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].name;
-                    curSearchString = curItr.value;
+                for ( let i = 0; i < curBench["metrics"].length; i++ ) {
 
+                    curMetric = curBench["metrics"][i].name;
+                    curSearchString = curItr.value;
                     // metric values will be found within the result of the outer regex
-                    if (BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex !== undefined) {
-                        if ( ( curRegexResult = BenchmarkMetric[curBenchmarkNamedRouted]["metrics"][i].outerRegex.exec( curSearchString ) ) !== null ) {
+                    if (curBench["metrics"][i].outerRegex !== undefined) {
+                        if ( ( curRegexResult = curBench["metrics"][i].outerRegex.exec( curSearchString ) ) !== null ) {
```

Comment:
Let's give an example here for cases with and without `outerRegex`. For example: If throughput is run and we don't use `outerRegex`, then we might get an array of size > 2.